### PR TITLE
BFD-3483: Simplify Migrator Configuration

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-3483](https://jira.cms.gov/browse/BFD-3483)

**User Story or Bug Summary:**
Simplify Migrator Configuration

---

### What Does This PR Do?
This simplifies the configuration strategy for the bfd-db-migrator in de-duplication of stored parameters and the migrator workload responsible for fetching its own configuration from the given context. The resultant startup script no longer contains sensitive values for path-to-production environments, but leaves these scripts flexible for operators to inject overrides as environment variables for debugging and development purposes.

Additionally, this removes a handful of unused migrator-related variables from both the terraform configuration and the Jenkins pipelines that provide them. These values haven't been relevant in months.

Finally, migrator deployments return to running only when necessary with a fix to a recently discovered bug. This fix yields the following snippet from the Jenkins deployment where the migrator is not necessary:
```
Comparing schema migration versions...
[Pipeline] sh
+ aws ssm get-parameter --name /bfd/test/common/nonsensitive/database_schema_version --region us-east-1 --with-decryption --query Parameter.Value --output text
[Pipeline] echo
Stored Schema Version : 20240606101236296
[Pipeline] sh
+ ./ops/jenkins/scripts/getLatestSchemaMigrationScriptVersion.sh
[Pipeline] echo
Latest Available Migration Version: 20240606101236296
[Pipeline] echo
Migrator deployment is NOT required. Skipping migrator deployment.
```

### How was this Evaluated
This was successfully deployed to `test` both through the Jenkins pipeline and out-of-band from a local development environment. Resultant scripts from templates are cleaner and clearer and the migrator functions correctly.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/services/base/values/prod-sbx.yaml [hunk 1]
```diff
@@ -8,8 +8,7 @@
...
-/bfd/${env}/migrator/sensitive/db_migrator_db_username: <<CIPHER>>AgV4vgdUgnDSkEDWE6mgfmHW_ln1JnXVbx4tnAhxZwPw15wAdAACAAtBcHBsaWNhdGlvbgAGY2lwaGVyABVhd3MtY3J5cHRvLXB1YmxpYy1rZXkAREEzU2RJV2lMLytWM215TTdNZmxHUWUybHZ5eGRHYUt3MUlkR1prMXE4SlVvNUJFa0ZwMGRZVXc2b1U3TWtXR2hoZz09AAEAB2F3cy1rbXMAS2Fybjphd3M6a21zOnVzLWVhc3QtMTo1NzczNzM4MzE3MTE6a2V5L21yay1jMjg5NTM3ODY2MzY0NTk4OTQ2NWQzMTJmYjcxMGU0NQC4AQIBAHjqKCGUsrfafVye_2mw4SSOkeKVkIdNH7fvxxo_a9cGpQETcXD4abz7JSrHmnc3Rt15AAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMSj5oYdTdqVMTCf6-AgEQgDtAizh7RVS7u6_t-4Xm5AT8HFIWm5bpynDnxAyxJD0kyDm_9iQnTQBdVnTHQhgrCL6Oys7QiwO-Hxeu9QIAABAANV2ZHdC4IG1ar6e1F2Wm_HIYkEq6AegkxmX1WR_Mcm9TDFrHByQd73HBnj6jo9i5_____wAAAAEAAAAAAAAAAAAAAAEAAAATR8HiIeY773hqrkXUvnNHrMLZBafYwxdt_NQdqBJi0s5B0dQAZzBlAjBGCyTqE9uXTVoGsmIM-wZ5vRa_ajybyleIEXnkdZFjXF0j32_JC4aOySHXJcDK55YCMQCTUK8ciY9xNBVyeEqtfF9UO2Je5v4BNH0mRheobCrEzVeiMBnhI4Gxus4NoBszAWk=<</CIPHER>>
```
Comment (aschey-forpeople) [inline ops/terraform/services/base/values/prod-sbx.yaml:12]: I think we may need to update `ephemeral.tf` to account for some of these removals.

Comment (mjburling) [reply ops/terraform/services/base/values/prod-sbx.yaml:12]: > I think we may need to update `ephemeral.tf` to account for some of these removals.

Thanks! Updated.

**Review by malessi — APPROVED** [review]
Nothing jumps out to me here. One step closer to 3045 being realized 😉.

**Review by aschey-forpeople — APPROVED** [review]
🚢

**Review by dondevun — APPROVED** [review]

**Pull request merged**